src/main/java/org/folio/rest/impl/UploadBatchVoucherExportHelper.java

```diff
@@ -149,8 +151,8 @@ private Future<Void> updateHolderWithBatchVoucher(BatchVoucherUploadHolder uploa
 
   private Future<Void> failUploadUpdate(BatchVoucherExport bvExport, Throwable t) {
     if (bvExport != null) {
-      if (!CREDENTIALS_NOT_FOUND.equals(t.getMessage())
-                      && !(t instanceof URISyntaxException)) {
+      if (Boolean.FALSE.equals(URL_NOT_FOUND_FOR_FTP.equals(t.getMessage()) || URI_SYNTAX_ERROR.equals(t.getMessage())
```

Comment:
This part actually like this `Boolean.FALSE.equals(URL_NOT_FOUND_FOR_FTP.equals(t.getMessage()) || URI_SYNTAX_ERROR.equals(t.getMessage()) || CREDENTIALS_NOT_FOUND.equals(t.getMessage()))`
if it will be changed to `if(!(URL_NOT_FOUND_FOR_FTP.equals(t.getMessage()) || URI_SYNTAX_ERROR.equals(t.getMessage()) || CREDENTIALS_NOT_FOUND.equals(t.getMessage()))))
`
I thought it will make code hard to read a little bit, so that is why Boolean.FALSE.equals is being used

Reply:
Got it, it's already very hard to read for me : )
We have a following semantics:
```
      if (
        !URL_NOT_FOUND_FOR_FTP.equals(message)
        && !URI_SYNTAX_ERROR.equals(message)
        && !CREDENTIALS_NOT_FOUND.equals(message)
      ) {
```
What you make hard to read is set of negations and conjunctions ( !a && !b && !c).
In this case you can have a variable `x = a || b || c` and use `!x`:
```
      boolean isKnownError =
        URL_NOT_FOUND_FOR_FTP.equals(message) || 
        URI_SYNTAX_ERROR.equals(message) ||
        CREDENTIALS_NOT_FOUND.equals(message);
      if (!isKnownError) {
        bvExport.setStatus(BatchVoucherExport.Status.ERROR);
      }
```
This also clarifies that we set error status when exception is not known or whatever variable name says.

Reply:
Thanks, code was improved